### DISCUSSION TITLE
`tsed-cli` with ESM

### DISCUSSION THREAD

**Discussion by joeynq** [TypedProject/ts-express-decorators#2350] [Q&A] [answered]
Hi @Romakita,

I'm having trouble working with the CLI to generate swagger (`GenerateSwaggerCmd`). This is for CI environment to generate schema and do stuffs.

The issue came from `oidc-provider@8.1.0` which is ESM. It's seems I can't make `ts-node` to work with ESM.

I have some local modules, in the same pnpm monorepo, and import `.ts` files directly from module without compiling, so changing `tsconfig.json` is a bit tricky for me.

My code using webpack/ts-loader to bundle and run the app, and `oidc-provider` doesn't have any issues with it.

I'm asking if anything I can do to make this work?

Thanks in advanced.

**Comment by Romakita** [comment, marked as answer]
Cli isn’t commletely compatible with ESM. CLI use the server to load controllers and generate swagger.json

I haven’t idea to help you this point. On my project it works because we don’t import oidc-provider using esm import but by lazy loading it.

**Reply by joeynq** [reply]
Thanks for your response!

I updated all `oidc-provider` import to lazy loading and it works.

BTW, do you think we should change `@tsed/oidc-provider` to use lazy import?

**Reply by Romakita** [reply]
@tsed/oidc-provider already lazy load the module:
https://github.com/tsedio/tsed/blob/production/packages/security/oidc-provider/src/services/OidcProviderNodeModule.ts

and the module is an injectable service:
https://github.com/tsedio/tsed/blob/production/packages/security/oidc-provider/src/services/OidcProvider.ts#L68 

see you ^^
Romain
